src/lib.rs

```diff
@@ -513,73 +513,127 @@ impl ColoredString {
     fn has_colors() -> bool {
         false
     }
+}
 
-    fn compute_style(&self) -> String {
-        if !Self::has_colors() || self.is_plain() {
-            return String::new();
+#[derive(Debug, Clone, Copy)]
+struct EscapeInnerResetSequencesHelper<'a> {
+    input: &'a str,
+    fgcolor: Option<Color>,
+    bgcolor: Option<Color>,
+    style: style::Style,
+    is_plain: bool,
+}
```

Comment:
`lib.rs` is getting pretty big. Can you move this to a separate module/file, e.g. `helpers.rs`, along with the related tests?

Also, even though these are private, and you add doc comments to these new types to describe what they're for?